keps/sig-windows/0000-20190103-windows-node-support.md

```diff
@@ -0,0 +1,112 @@
+         * [What will work eventually](#what-will-work-eventually)
+         * [What will never work (without underlying OS changes)](#what-will-never-work-without-underlying-os-changes)
+         * [Relevant resources/conversations](#relevant-resourcesconversations)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+      * [Other references](#other-references)
+
+
+## Summary
+
+There is strong interest in the community for adding support for workloads running on Microsoft Windows. This is non-trivial due to the significant differences in the implementation of Windows from the Linux-based OSes that have so far been supported by Kubernetes.
+
+
+## Motivation
+
+Windows-native workloads still account for a significant portion of the enterprise software space. While containerization technologies emerged first in the UNIX ecosystem, Microsoft has made investments in recent years to enable support for containers in its Windows OS. As users of Windows increasingly turn to containers as the preferred abstraction for running software, the Kubernetes ecosystem stands to benefit by becoming a cross-platform cluster manager.
+
+### Goals
+
+- Enable users to run nodes on Windows servers 
+- Document the differences and limitations compared to Linux
+- Test results added to testgrid to prevent regression of functionality 
+
+### Non-Goals
+
+- Adding Windows support to all projects in the Kubernetes ecosystem (Cluster Lifecycle, etc)
+
+## Proposal
+
+As of 29-11-2018 much of the work for enabling Windows nodes has already been completed. Both `kubelet` and `kube-proxy` have been adapted to work on Windows Server, and so the first goal of this KEP is largely already complete. 
+
+### What works today
+- Windows-based containers can be created by kubelet, [provided the host OS version matches the container base image](https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility)
+    - ConfigMap, Secrets: as environment variables or  volumes
+    - Resource limits
+    - Pod & container metrics
+- Pod networking with [Azure-CNI](https://github.com/Azure/azure-container-networking/blob/master/docs/cni.md), [OVN-Kubernetes](https://github.com/openvswitch/ovn-kubernetes), [two CNI meta-plugins](https://github.com/containernetworking/plugins), [Flannel](https://github.com/coreos/flannel) and [Calico](https://github.com/projectcalico/calico)
+- Dockershim CRI
+- Many<sup id="a1">[1]</sup> of the e2e conformance tests when run with [alternate Windows-based images](https://hub.docker.com/r/e2eteam/) which are being moved to [kubernetes-sigs/windows-testing](https://www.github.com/kubernetes-sigs/windows-testing)
+- Persistent storage: FlexVolume with [SMB + iSCSI](https://github.com/Microsoft/K8s-Storage-Plugins/tree/master/flexvolume/windows), and in-tree AzureFile and AzureDisk providers
+ 
```

Comment:
Do pod hostname and subdomain fields work? How about hostAliases? dnsConfig?

Reply:
Basically I expect someone to read through PodSpec field by field to make sure we haven't forgotten something.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2743